flake.nix

```diff
@@ -14,56 +14,97 @@
+            # hash = "sha256-FtJFZorlGqPBfkPgFbEztNvYHweFaRVeuAM8xOMleMk=";
+            # NOTE
+            rev = "80a243e07ef77156af66efa7d22ac35aba44c1b3"; # ECTester submodule version at the time of writing
+            hash = "sha256-Sa1XjU7wi4umVQ6BUj9BxJMHYlXNg6xw9Cb/vBE+ScQ=";
           };
-          # FIXME this might cause unwanted things
-          patches = [];
+
+          # NOTE this build does not match upstream, but is what ECTester did at the time of writing
+          buildPhase = ''
+            cmake -GNinja -DBUILD_SHARED_LIBS=1 -Bbuild
+            pushd build
+            ninja crypto
+            popd
+          '';
+
+          installPhase = ''
+            mkdir --parents $bin/bin $dev $out/lib
+            mv include $dev
+
+            pushd build
+            mv crypto/libcrypto.so $out/lib/lib_boringssl.so
+            popd
+          '';
+
         });
-        openssl_315 = pkgs.openssl.overrideAttrs (_old: rec {
-          version = "3.1.5";
-          pname = "openssl";
-          src = pkgs.fetchurl {
-            url = "https://www.openssl.org/source/openssl-${version}.tar.gz";
-            hash = "sha256-auAVRn2r8EabE5rakzGTJ74kuYJR/67O2gIhhI3AkmI=";
-          };
-          # FIXME this might cause unwanted things
-          patches = [];
+
+
+        libressl = pkgs.libressl.overrideAttrs (_old: rec {
+          # devLibPath = pkgs.lib.makeLibraryPath [ pkgs.libressl.dev ];
+          # pname = "libressl";
+          # version = "3.9.2";
+          # includes = [ "tests/tlstest.sh" ];
+          # src = pkgs.fetchurl {
+          #   url = "mirror://openbsd/LibreSSL/${pname}-${version}.tar.gz";
+          #   hash = "sha256-ewMdrGSlnrbuMwT3/7ddrTOrjJ0nnIR/ksifuEYGj5c=";
+          # };
+          # nativeBuildInputs = _old.nativeBuildInputs ++ (with pkgs; [
+          #   pkg-config
+          # ]);
+
+          # Patched according to the previous versions:
+          # https://github.com/NixOS/nixpkgs/blob/nixos-24.05/pkgs/development/libraries/libressl/default.nix#L118
+          # For unknown reasons the newer versions are not patched this way (yet?)
+          patches = [
+            (pkgs.fetchpatch {
+              url = "https://github.com/libressl/portable/commit/86e4965d7f20c3a6afc41d95590c9f6abb4fe788.patch";
+              includes = [ "tests/tlstest.sh" ];
+              hash = "sha256-XmmKTvP6+QaWxyGFCX6/gDfME9GqBWSx4X8RH8QbDXA=";
+            })
+          ];
+
+          # NOTE: Due to name conflicts between OpenSSL and LibreSSL we need to resolve this manually.
+          postFixup =  pkgs.lib.concatLines [ 
+            ( _old.postFixup or "" )
+            ''
+            cp $dev/lib/pkgconfig/libcrypto.pc $dev/lib/pkgconfig/libresslcrypto.pc
+            sed --in-place --expression 's/-lcrypto/-lresslcrypto/' $dev/lib/pkgconfig/libresslcrypto.pc
+            ln -s $out/lib/libcrypto.so $out/lib/libresslcrypto.so
```

Comment:
patchelf necessary here. The library knows its soname, see:
```
readelf -d  libresslcrypto.so | grep SONAME
```
Will print out `libcrypto.so` which is wrong.

Reply:
The .so is now not necessary anymore right?